### PR TITLE
[spirv] Fix ConstantBuffer/TextureBuffer assignment

### PR DESCRIPTION
**Pull request by antiagainst**
Assigning a ConstantBuffer<T> to a T is indeed a copy.

For addressing #838.

### CODE REVIEW

**Comment by antiagainst** [comment]
This is the 3rd PR for fixing buffer aliasing. Only the last commit needs review. The others are already in #918 & #923.

File: tools/clang/lib/SPIRV/SPIRVEmitter.cpp [hunk 1]
```diff
@@ -199,21 +199,6 @@ const DeclContext *isConstantTextureBufferDeclRef(const Expr *expr) {
   return nullptr;
 }
 
-/// Returns true if the given expr is loading a ConstantBuffer/TextureBuffer
-/// as a whole.
-bool isConstantTextureBufferLoad(const Expr *expr) {
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/SPIRVEmitter.cpp]: yay!

File: tools/clang/lib/SPIRV/SPIRVEmitter.h [hunk 2]
```diff
@@ -106,6 +106,11 @@ class SPIRVEmitter : public ASTConsumer {
   SpirvEvalInfo doMemberExpr(const MemberExpr *expr);
   SpirvEvalInfo doUnaryOperator(const UnaryOperator *expr);
 
+  /// Overload with pre computed SpirvEvalInfo.
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/SPIRVEmitter.h]: Nit: would you put these two next to each other?

File: tools/clang/test/CodeGenSPIRV/spirv.legal.tbuffer.hlsl [hunk 3]
```diff
@@ -19,8 +19,8 @@ struct S {
     float4 f;
 };
 
-// CHECK: %myCBuffer = OpVariable %_ptr_Uniform_type_TextureBuffer_S Uniform
-TextureBuffer<S>         myCBuffer;
+// CHECK: %myTBuffer = OpVariable %_ptr_Uniform_type_TextureBuffer_S Uniform
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/spirv.legal.tbuffer.hlsl]: lol, I should have caught this in the previous reviews! oops

Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/spirv.legal.tbuffer.hlsl]: Haha. No worries. :)

File: tools/clang/lib/SPIRV/SPIRVEmitter.h (hunk 2 above)
Comment (antiagainst) [reply tools/clang/lib/SPIRV/SPIRVEmitter.h]: This one is a `private:` one. The other one is `public:`.

Comment (ehsannas) [reply tools/clang/lib/SPIRV/SPIRVEmitter.h]: Oh sorry. yes. Never mind then. Thanks

**Comment by antiagainst** [comment]
Waiting on KhronosGroup/SPIRV-Tools#1096 to submit

**Pull request merged**